flow/connectors/mongo/mongo.go

```diff
@@ -216,7 +222,9 @@ func (c *MongoConnector) PullRecords(
 	changeStreamOpts := options.ChangeStream().
 		SetComment("PeerDB changeStream for mirror " + req.FlowJobName).
 		SetFullDocument(options.UpdateLookup).
-		SetFullDocumentBeforeChange(options.Off)
+		SetFullDocumentBeforeChange(options.Off).
+		SetMaxAwaitTime(req.IdleTimeout)
```

Comment:
Reading: https://github.com/mongodb/specifications/blob/master/source/change-streams/change-streams.md#why-do-we-need-to-expose-the-postbatchresumetoken

IIUC, this setting should help us resolve the stale resume token when there are not events. when Max Await Time is reached and we haven't received a record, `Next`  should return false (but without errors) and the latest resumeToken from `GetMore` should get populated which we can store.